lib/hotbar.js

```diff
@@ -19,40 +44,112 @@ class HotBar extends LitElement {
+      slotImage.innerHTML = ''
+      slotImage.appendChild(await icon)
+
+      slotImage.nextElementSibling.innerHTML = newItem?.count > 1 ? newItem.count : ''
     })
   }
 
   async reloadHotbar () {
     for (let i = 0; i < 9; i++) {
       const item = this.bot.inventory.slots[this.bot.inventory.hotbarStart + i]
       const sprite = item ? invsprite[item.name] : invsprite.air
-      const slotImage = this.shadowRoot.getElementById('hotbar-' + i)
-      slotImage.style['background-position-x'] = `-${sprite.x * 2}px`
-      slotImage.style['background-position-y'] = `-${sprite.y * 2}px`
-      slotImage.innerHTML = item?.count > 1 ? item.count : ''
+      const slotImage = this.shadowRoot.getElementById('hotbar-' + i).firstElementChild
+
+      const icon = await getItemIcon(sprite)
+      slotImage.innerHTML = ''
+      slotImage.appendChild(await icon)
+
+      slotImage.nextElementSibling.innerHTML = item?.count > 1 ? item.count : ''
     }
   }
 
   async reloadHotbarSelected (slot) {
     const item = this.bot.inventory.slots[this.bot.inventory.hotbarStart + slot]
-    const planned = (20 * 4 * slot) + 'px'
-    this.shadowRoot.getElementById('hotbar-highlight').style.marginLeft = planned
+
+    const hotbarSlots = this.shadowRoot.querySelectorAll('.hotbar-slot')
+    for (let i = 0; i < hotbarSlots.length; i++) {
+      const hotbarSlot = hotbarSlots[i]
+
+      hotbarSlot.classList.remove('selected')
+      if (slot === i) hotbarSlot.classList.add('selected')
+    }
+
     this.bot.setQuickBarSlot(slot)
     this.activeItemName = item?.displayName
     const name = this.shadowRoot.getElementById('hotbar-item-name')
     name.classList.remove('hotbar-item-name-fader')
     setTimeout(() => name.classList.add('hotbar-item-name-fader'), 10)
   }
 
+  async reloadHealth (health) {
+    const renderer = this.shadowRoot.getElementById('health-renderer')
+    if (this.bot.game.gameMode === 'creative') renderer.style.display = 'none'
+    else renderer.style.display = 'block'
+
+    const rows = Math.ceil(health / 20)
+    const hearts = Math.floor(health / 2)
+    const half = health % 2 !== 0
+
+    renderer.innerHTML = ''
+    for (let i = 0; i < rows; i++) {
+      const row = document.createElement('div')
+      row.className = 'heart-row'
+
+      for (let j = 0; j < 10; j++) {
+        const icon = document.createElement('div')
+        icon.classList.add('heart-icon')
+
+        if (this.bot.game.hardcore) icon.classList.add('hardcore')
+        if (j < hearts) icon.classList.add('full')
+        else if (half && j === hearts) icon.classList.add('half')
+
+        row.appendChild(icon)
+      }
+
+      renderer.appendChild(row)
+    }
+  }
+
+  async reloadFood (food) {
+    const foods = Math.floor(food / 2)
+    const half = food % 2 !== 0
+
+    const renderer = this.shadowRoot.getElementById('food-renderer')
+    if (this.bot.game.gameMode === 'creative') renderer.style.display = 'none'
```

Comment:
What about spectator mode? I don't see any code to check for that.

Reply:
I haven't made the spectator mode hotbar so idk what I would add that check.

Reply:
I meant for hiding the health at least.

Reply:
Fixed by commit 15caaea.